core/src/main/java/com/linecorp/armeria/internal/common/util/TemporaryThreadLocals.java

```diff
@@ -15,82 +15,103 @@
  */
 package com.linecorp.armeria.internal.common.util;
 
+import static com.google.common.base.Preconditions.checkState;
+
 import com.google.common.annotations.VisibleForTesting;
+import com.google.errorprone.annotations.MustBeClosed;
 
 import io.netty.util.internal.EmptyArrays;
 
 /**
  * Provides various thread-local variables used by Armeria internally, mostly to avoid allocating
  * short-living objects, such as {@code byte[]} and {@link StringBuilder}. Keep in mind that the variables
- * provided by this class must be used with extreme care, because otherwise it will result in unpredictable
- * behavior.
+ * provided by this class must be used and released with extreme care, because otherwise it will result in
+ * unpredictable behavior.
  *
  * <p>Most common mistake is to call or recurse info a method that uses the same thread-local variable.
  * For example, the following code will produce a garbled string:
```

Comment:
```suggestion
 * For example, the following code will throw an exception because {@link TemporaryThreadLocals#acquire()} is called twice before the first call is closed:
```

Reply:
I would like to keep showing the specific case of the side effect, but it make the explanation verbose. Agreed to update.